cyclonedx/cyclonedx.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Sonatype Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Definitions and functions for processing golang purls into a CycloneDX Sbom
+package cyclonedx
+
+import (
+	"encoding/xml"
+	"fmt"
+	"strings"
+)
+
+type Sbom struct {
+	XMLName    xml.Name   `xml:"bom"`
+	Xmlns      string     `xml:"xmlns,attr"`
+	Version    string     `xml:"version,attr"`
+	Components Components `xml:"components"`
+}
+
+type Components struct {
+	Component []Component `xml:"component"`
+}
+
+type Component struct {
+	Type    string `xml:"type,attr"`
+	BomRef  string `xml:"bom-ref,attr"`
+	Name    string `xml:"name"`
+	Version string `xml:"version"`
+	Purl    string `xml:"purl"`
```

Comment:
I feel like all of this should go into `types.go`, any opinions?

Reply:
Yeah i think that would make sense

Reply:
I'll handle that in the AM.

Reply:
Types I wanted public moved in: https://github.com/sonatype-nexus-community/nancy/pull/68/commits/4adf87f2bc642b09ff7b42208f1209a1b78cfa54

Made SBOM types public (because I'll eventually create a method to return a struct rather than a string, so people can do more with it).

Reply:
Decided against struct for now because golang and xml unmarshall is actually kinda....bad! Loses some xmlns stuff, blew my mind. String was a way to guarantee nothing was lost.